backend/pkg/models/database/generate.go

```diff
@@ -47,6 +51,22 @@ PLEASE DO NOT EDIT BY HAND
 `, "\n"), "\n")
 
 func main() {
+	// Read config file for database type
+	appconfig, err := config.Create()
```

Comment:
Compile time vs runtime. 
The user's configuration file will not be present during compilation time, so the database type choice has not been made yet. 
We need to generate both Sqlite & Postgres compatible structs (in different directories).
The generated `utils.NewFhirResourceModelByType` fn should take in the DatabaseRepositoryType as a parameter, and conditionally point to a different set of generated structs.